src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -47,143 +46,331 @@ namespace winrt::TerminalApp::implementation
         switch (action)
         {
         case ShortcutAction::CopyText:
-            _CopyTextHandlers(true);
-            return true;
+        {
+            CopyTextArgs args{};
+            args.TrimWhitespace(true);
+            ActionEventArgs eventArgs{ args };
+            _CopyTextHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
         case ShortcutAction::CopyTextWithoutNewlines:
-            _CopyTextHandlers(false);
-            return true;
+        {
+            CopyTextArgs args{};
+            args.TrimWhitespace(false);
+            ActionEventArgs eventArgs{ args };
+            _CopyTextHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
         case ShortcutAction::PasteText:
-            _PasteTextHandlers();
-            return true;
+        {
+            ActionEventArgs eventArgs{};
+            _PasteTextHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
         case ShortcutAction::NewTab:
-            _NewTabHandlers();
-            return true;
+        {
+            ActionEventArgs eventArgs{};
+            _NewTabHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
         case ShortcutAction::DuplicateTab:
-            _DuplicateTabHandlers();
-            return true;
+        {
+            ActionEventArgs eventArgs{};
+            _DuplicateTabHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
         case ShortcutAction::OpenSettings:
-            _OpenSettingsHandlers();
-            return true;
+        {
+            ActionEventArgs eventArgs{};
+            _OpenSettingsHandlers(*this, eventArgs);
+            return eventArgs.Handled();
+        }
 
         case ShortcutAction::NewTabProfile0:

```

Comment:
So I don't think this will work in the general case - right now there's a few actions with a single arg, but the macro would quickly get M A D once there were multiple arguments to an action. PLus there's other work coming here with pt 2 of #1349, so I think I'm going to leave it like this _for now_ and keep that in mind for the upcoming feature work here